Add baseDir to babel plugin to enable caching

mydea opened the pull request:
The build under embroider complained that this plugin prevented proper caching. I've tried it in our app and defining the `baseDir` made the warning go away.

Fixes https://github.com/tildeio/ember-element-helper/issues/53

mydea commented:
Any chance to get this merged? Would be nice to get rid of this warning :)

rwjblue reviewed (approved):
Seems good to me! Thank you!

rwjblue reviewed (changes requested):
```$ eslint .

/home/runner/work/ember-element-helper/ember-element-helper/lib/element-helper-syntax-plugin.js
Error:   43:2  error  Unnecessary semicolon  no-extra-semi

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

mydea commented:
> ```
> 
> /home/runner/work/ember-element-helper/ember-element-helper/lib/element-helper-syntax-plugin.js
> Error:   43:2  error  Unnecessary semicolon  no-extra-semi
> 
> ✖ 1 problem (1 error, 0 warnings)
>   1 error and 0 warnings potentially fixable with the `--fix` option.
> ```

Oops, fixed it, rebased on main and pushed again!
